Add a README for the content directory (#693)
discord-python/site: pydis_site/apps/content/README.md

# The "content" app

This application serves static, Markdown-based content. Django-wise there is
relatively little code in there; most of it is concerned with serving our
content.

## Contributing pages

The Markdown files hosting our content can be found in the
[`resources/`](./resources) directory. The process of contributing to pages is
covered extensively in our online guide which you can find
[here](https://www.pythondiscord.com/pages/guides/pydis-guides/how-to-contribute-a-page/).
Alternatively, read it directly at
[`resources/guides/pydis-guides/how-to-contribute-a-page.md`](./resources/guides/pydis-guides/how-to-contribute-a-page.md).

## Directory structure

Let's look at the structure in here:

- `resources` contains the static Markdown files that make up our site's
  [pages](https://www.pythondiscord.com/pages/)

- `tests` contains unit tests for verifying that the app works properly.

> - `views` contains Django views which generating and serve the pages from the

Reviewer: ```suggestion
- `views` contains Django views which generate and serve the pages from the
```

Author: Sorry Mark, I have the brain of a parrot. Rest assured this mistake is fixed and won't happen again in the future.

Thanks for the review!

# The "content" app

> This application serves static, markdown-based content. Django-wise there is

Reviewer: All references to "Markdown" should be capitalised.

Author: ca2c583b15d46d7821a03ec047678429f96c7e03

> relatively little code in there, most of it is concerned with serving our

Reviewer: This is a comma splice.
```suggestion
relatively little code in there; most of it is concerned with serving our
```

Author: ca2c583b15d46d7821a03ec047678429f96c7e03

content.

The markdown files hosting our content can be found in the

> [`resources/`](./resources) directory, and contain

Reviewer: Incomplete sentence?

Author: ca2c583b15d46d7821a03ec047678429f96c7e03

## Contributing pages

> Contributing pages is covered extensively in our online guide, which you can
> find

Reviewer: ```suggestion
The process of contributing to pages is covered extensively in our online guide, which you can
find
```

Author: ca2c583b15d46d7821a03ec047678429f96c7e03

[here](https://www.pythondiscord.com/pages/guides/pydis-guides/how-to-contribute-a-page/)

> or alternatively read it directly at

Reviewer: Split the sentence up so it doesn't get so long.
```suggestion
Alternatively, read it directly at
```

Author: ca2c583b15d46d7821a03ec047678429f96c7e03

[`resources/guides/pydis-guides/how-to-contribute-a-page.md`](./resources/guides/pydis-guides/how-to-contribute-a-page.md).

## Directory structure

Let's look at the structure in here:

- `resources` contains the static Markdown files that make up our site's
  [pages](https://www.pythondiscord.com/pages/)

- `migrations` contains standard Django migrations. As the `content` app

>   contains purely static markdown files, no migrations are present here.

Reviewer: Why do we even keep this directory around then? Will Django complain if it can't find it?

Author: Good point. I've removed it in 4ad92b98a0df047b12e302ce8d849c2bd85862f1.

- `tests` contains unit tests for verifying that the app works properly.

> - `views` contains our Django views generating and serving the pages from the

Reviewer: ```suggestion
- `views` contains Django views which generate and serve the pages from the
```

Author: ca2c583b15d46d7821a03ec047678429f96c7e03

  input markdown.

As for the modules, apart from the standard Django modules in here, the
`utils.py` module contains utility functions for discovering markdown files to

> serve on the filesystem.

Reviewer: I don't think this clarification is necessary, and it introduces ambiguity: are files being served on the filesystem, or are files, which are on the filesystem, being served? Of course the former doesn't even make sense, but it's confusing regardless.

```suggestion
serve.
```

Author: ca2c583b15d46d7821a03ec047678429f96c7e03
